fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biomes/InternalBiomeData.java

```diff
@@ -0,0 +1,183 @@
+		OVERWORLD_INJECTED_BIOMES.add(hills);
+	}
+
+	public static void addOverworldShoreBiome(Biome parent, Biome shore, double weight) {
+		Preconditions.checkArgument(parent != null && shore != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_SHORE_MAP.computeIfAbsent(parent, biome -> new WeightedBiomePicker()).addBiome(shore, weight);
+		OVERWORLD_INJECTED_BIOMES.add(shore);
+	}
+
+	public static void addOverworldEdgeBiome(Biome parent, Biome edge, double weight) {
+		Preconditions.checkArgument(parent != null && edge != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_EDGE_MAP.computeIfAbsent(parent, biome -> new WeightedBiomePicker()).addBiome(edge, weight);
+		OVERWORLD_INJECTED_BIOMES.add(edge);
+	}
+
+	public static void addOverworldBiomeReplacement(Biome replaced, Biome variant, double chance, OverworldClimate[] climates) {
+		Preconditions.checkArgument(replaced != null && variant != null, "One or both arguments are null");
+		Preconditions.checkArgument(chance > 0 && chance <= 1, "Chance is not greater than 0 or less than or equal to 1");
+		OVERWORLD_VARIANT_TRANSFORMERS.computeIfAbsent(replaced, biome -> new VariantTransformer()).addBiome(variant, chance, climates);
+		OVERWORLD_INJECTED_BIOMES.add(variant);
+	}
+
+	public static void setOverworldRiverBiome(Biome parent, Biome river) {
+		OVERWORLD_RIVER_MAP.put(parent, river);
+		if (river != null) {
+			OVERWORLD_INJECTED_BIOMES.add(river);
+		}
+	}
+
+	public static void addSpawnBiome(Biome biome) {
+		SPAWN_BIOMES.add(biome);
+	}
+
+	public static List<Biome> getOverworldInjectedBiomes() {
+		return OVERWORLD_INJECTED_BIOMES;
+	}
+
+	public static Set<Biome> getSpawnBiomes() {
+		return SPAWN_BIOMES;
+	}
+
+	public static Map<Biome, WeightedBiomePicker> getOverworldHills() {
+		return OVERWORLD_HILLS_MAP;
+	}
+
+	public static Map<Biome, WeightedBiomePicker> getOverworldShores() {
+		return OVERWORLD_SHORE_MAP;
+	}
+
+	public static Map<Biome, WeightedBiomePicker> getOverworldEdges() {
+		return OVERWORLD_EDGE_MAP;
+	}
+
+	public static Map<Biome, Biome> getOverworldRivers() {
+		return OVERWORLD_RIVER_MAP;
+	}
+
+	public static EnumMap<OverworldClimate, WeightedBiomePicker> getOverworldBaseModdedBiomePickers() {
+		return OVERWORLD_BASE_MODDED_BIOME_PICKERS;
+	}
+
+	public static Map<Biome, VariantTransformer> getOverworldVariantTransformers() {
+		return OVERWORLD_VARIANT_TRANSFORMERS;
+	}
+
+	private static class DefaultHillsData {
+		private static final HashMap<Biome, Biome> DEFAULT_HILLS = new HashMap<>();
```

Comment:
Why is this not immutable?

Reply:
I suggest using `Map<Biome, Biome>` for field type.

Reply:
DefaultHillsData was written by @coderbot16 afaik. I would have used `Map<Biome, Biome>` myself as @liach said but making it immutable sounds like a good suggestion

Reply:
It's a private field so wouldn't immutability and type not really matter here? This map is completely internal to the class and is not accessible at all from the outside.

Reply:
Immutability is better for performance and thread safety.

Reply:
Good coding practices always matter ! Using a mutable map is not too big of a deal, but using the broadest type should always be done.